src/hypothesis/internal/compat.py

```diff
@@ -290,16 +290,8 @@ def inner(*args, **kwargs):
         getfullargspec = silence_warnings(getfullargspec)
 
 
-def get_type_hints(thing):
-    """Try to return any type hints for ``thing``."""
-    try:
-        import typing
-        return typing.get_type_hints(thing)
```

Comment:
That was the first thing I tried, and it ran into bunch of additional issues.

I could add fallback cases for those too, but my reasoning is that either the fallback method works well enough for our purposes (in which case we might as well just use it) or having it only trigger in weird edge cases will just make the bugs we run into more obscure and more annoying to track down.